CI: Runtime Test (privileged) - Test_NeighborAddDel

## CI failure

Hit in https://github.com/cilium/cilium/pull/25966: https://github.com/cilium/cilium/actions/runs/5197157864/jobs/9371726504?pr=25966

```
2023-06-07T07:36:34.2011115Z === RUN   Test_NeighborAddDel
2023-06-07T07:36:34.2011575Z time="2023-06-07T07:35:06Z" level=info msg="deleting dummy links"
2023-06-07T07:36:34.2012097Z time="2023-06-07T07:35:06Z" level=info msg="adding dummy links"
2023-06-07T07:36:34.2012627Z time="2023-06-07T07:35:06Z" level=info msg="GoBGP test instance: starting"
2023-06-07T07:36:34.2013352Z time="2023-06-07T07:35:06Z" level=info msg="Add a peer configuration" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2014301Z time="2023-06-07T07:35:06Z" level=debug msg="IdleHoldTimer expired" Duration=0 Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2014948Z time="2023-06-07T07:35:06Z" level=info msg="GoBGP test instance: starting"
2023-06-07T07:36:34.2015648Z time="2023-06-07T07:35:06Z" level=info msg="Add a peer configuration" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2016464Z time="2023-06-07T07:35:06Z" level=debug msg="IdleHoldTimer expired" Duration=0 Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2017535Z time="2023-06-07T07:35:06Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_ACTIVE old=BGP_FSM_IDLE reason=idle-hold-timer-expired subsys=basic
2023-06-07T07:36:34.2018532Z level=info msg=Invoked duration="155.43µs" function="test.newFixture.func6 (fixtures.go:159)" subsys=hive
2023-06-07T07:36:34.2018986Z level=info msg=Starting subsys=hive
2023-06-07T07:36:34.2019743Z level=info msg="Start hook executed" duration="1.803µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1.Service].Start" subsys=hive
2023-06-07T07:36:34.2020734Z time="2023-06-07T07:35:06Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_ACTIVE old=BGP_FSM_IDLE reason=idle-hold-timer-expired subsys=basic
2023-06-07T07:36:34.2021499Z level=info msg="Start hook executed" duration=100.531909ms function="*manager.diffStore[*github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1.Service].Start" subsys=hive
2023-06-07T07:36:34.2022436Z level=info msg="Start hook executed" duration="2.244µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s.Endpoints].Start" subsys=hive
2023-06-07T07:36:34.2022965Z level=info msg="Using v1.Endpoints" subsys=k8s
2023-06-07T07:36:34.2023519Z level=info msg="Start hook executed" duration=100.794548ms function="*manager.diffStore[*github.com/cilium/cilium/pkg/k8s.Endpoints].Start" subsys=hive
2023-06-07T07:36:34.2024529Z level=info msg="Start hook executed" duration="1.823µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1.CiliumBGPPeeringPolicy].Start" subsys=hive
2023-06-07T07:36:34.2025508Z level=info msg="Start hook executed" duration="1.342µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1.Node].Start" subsys=hive
2023-06-07T07:36:34.2026447Z level=info msg="Start hook executed" duration="1.092µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2.CiliumNode].Start" subsys=hive
2023-06-07T07:36:34.2027285Z level=info msg="Start hook executed" duration="104.475µs" function="*agent.kubernetesNodeSpecer.Start" subsys=hive
2023-06-07T07:36:34.2027880Z level=info msg="Start hook executed" duration=100.555129ms function="*agent.Controller.Start" subsys=hive
2023-06-07T07:36:34.2028351Z === RUN   Test_NeighborAddDel/add_two_neighbors
2023-06-07T07:36:34.2028888Z time="2023-06-07T07:35:07Z" level=info msg="GoBGP test instance: Peer Event: {0 IDLE}"
2023-06-07T07:36:34.2029582Z level=info msg="Cilium BGP Control Plane Controller now running..." component=Controller.Run subsys=bgp-control-plane
2023-06-07T07:36:34.2030398Z level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
2023-06-07T07:36:34.2031243Z level=info msg="Registering BGP servers for policy with local ASN 65001" component=manager.registerBGPServer subsys=bgp-control-plane
2023-06-07T07:36:34.2032169Z level=info msg="Reconciling peers for virtual router with local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2033026Z level=info msg="Adding peer 172.16.100.2/32 65011 to local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2034035Z level=info msg="Add a peer configuration" Key=172.16.100.2 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2034923Z level=info msg="Done reconciling peers for virtual router with local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2035831Z level=info msg="Successfully registered GoBGP servers for policy with local ASN 65001" component=manager.registerBGPServer subsys=bgp-control-plane
2023-06-07T07:36:34.2036544Z time="2023-06-07T07:35:07Z" level=info msg="GoBGP test instance: Peer Event: {65001 IDLE}"
2023-06-07T07:36:34.2037138Z time="2023-06-07T07:35:07Z" level=info msg="GoBGP test instance: Peer Event: {65001 ACTIVE}"
2023-06-07T07:36:34.2037985Z level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
2023-06-07T07:36:34.2038860Z level=info msg="Reconciling peers for virtual router with local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2039697Z level=info msg="Adding peer 172.16.100.3/32 65012 to local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2040983Z level=info msg="Add a peer configuration" Key=172.16.100.3 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2041852Z level=info msg="Updating peer 172.16.100.2/32 65011 in local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2042703Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011} state:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011 session_state:IDLE router_id:\"<nil>\"} transport:{local_address:\"<nil>\"}}"
2023-06-07T07:36:34.2043614Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011} state:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011 session_state:ACTIVE router_id:\"<nil>\"} transport:{local_address:\"<nil>\"}}"
2023-06-07T07:36:34.2044512Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012} state:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012 session_state:IDLE router_id:\"<nil>\"} transport:{local_address:\"<nil>\"}}"
2023-06-07T07:36:34.2045416Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012} state:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012 session_state:ACTIVE router_id:\"<nil>\"} transport:{local_address:\"<nil>\"}}"
2023-06-07T07:36:34.2046442Z level=info msg="Update timer configuration" Err="<nil>" Key=172.16.100.2 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2047196Z level=info msg="Resetting peer 172.16.100.2 (ASN 65011) due to a config change" subsys=bgp-control-plane
2023-06-07T07:36:34.2048060Z level=info msg="Send operation notification" Key=172.16.100.2 Op="Neighbor reset" Topic=Operation asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2048991Z level=info msg="Done reconciling peers for virtual router with local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2049819Z time="2023-06-07T07:35:13Z" level=debug msg="try to connect" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2050732Z time="2023-06-07T07:35:13Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_OPENSENT old=BGP_FSM_ACTIVE reason=new-connection subsys=basic
2023-06-07T07:36:34.2051476Z time="2023-06-07T07:35:13Z" level=info msg="GoBGP test instance: Peer Event: {65001 OPENSENT}"
2023-06-07T07:36:34.2052254Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011} state:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011 session_state:OPENSENT router_id:\"<nil>\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:46807}}"
2023-06-07T07:36:34.2053586Z time="2023-06-07T07:35:13Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_OPENCONFIRM old=BGP_FSM_OPENSENT reason=open-msg-received subsys=basic
2023-06-07T07:36:34.2054362Z time="2023-06-07T07:35:13Z" level=info msg="GoBGP test instance: Peer Event: {65001 OPENCONFIRM}"
2023-06-07T07:36:34.2055234Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011} state:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011 session_state:OPENCONFIRM router_id:\"172.16.100.2\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:46807}}"
2023-06-07T07:36:34.2056264Z time="2023-06-07T07:35:13Z" level=info msg="Peer Up" Key=172.16.100.1 State=BGP_FSM_OPENCONFIRM Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2057267Z time="2023-06-07T07:35:13Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_ESTABLISHED old=BGP_FSM_OPENCONFIRM reason=open-msg-negotiated subsys=basic
2023-06-07T07:36:34.2058346Z level=info msg="Peer Up" Key=172.16.100.2 State=BGP_FSM_OPENCONFIRM Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2059572Z time="2023-06-07T07:35:13Z" level=info msg="GoBGP test instance: Peer Event: {65001 ESTABLISHED}"
2023-06-07T07:36:34.2060195Z time="2023-06-07T07:35:13Z" level=info msg="GoBGP test instance: Peer Event: {0 IDLE}"
2023-06-07T07:36:34.2060977Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011} state:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011 session_state:ESTABLISHED router_id:\"172.16.100.2\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:46807}}"
2023-06-07T07:36:34.2061874Z time="2023-06-07T07:35:13Z" level=info msg="GoBGP test instance: Peer Event: {65001 IDLE}"
2023-06-07T07:36:34.2062497Z time="2023-06-07T07:35:13Z" level=info msg="GoBGP test instance: Peer Event: {65001 ACTIVE}"
2023-06-07T07:36:34.2063202Z time="2023-06-07T07:35:14Z" level=debug msg="try to connect" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2064097Z time="2023-06-07T07:35:14Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_OPENSENT old=BGP_FSM_ACTIVE reason=new-connection subsys=basic
2023-06-07T07:36:34.2064849Z time="2023-06-07T07:35:14Z" level=info msg="GoBGP test instance: Peer Event: {65001 OPENSENT}"
2023-06-07T07:36:34.2065611Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012} state:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012 session_state:OPENSENT router_id:\"<nil>\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:33577}}"
2023-06-07T07:36:34.2066785Z time="2023-06-07T07:35:14Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_OPENCONFIRM old=BGP_FSM_OPENSENT reason=open-msg-received subsys=basic
2023-06-07T07:36:34.2067560Z time="2023-06-07T07:35:14Z" level=info msg="GoBGP test instance: Peer Event: {65001 OPENCONFIRM}"
2023-06-07T07:36:34.2068327Z time="2023-06-07T07:35:14Z" level=info msg="Peer Up" Key=172.16.100.1 State=BGP_FSM_OPENCONFIRM Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2069184Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012} state:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012 session_state:OPENCONFIRM router_id:\"172.16.100.2\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:33577}}"
2023-06-07T07:36:34.2070339Z time="2023-06-07T07:35:14Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_ESTABLISHED old=BGP_FSM_OPENCONFIRM reason=open-msg-negotiated subsys=basic
2023-06-07T07:36:34.2071364Z time="2023-06-07T07:35:14Z" level=info msg="GoBGP test instance: Peer Event: {65001 ESTABLISHED}"
2023-06-07T07:36:34.2072199Z level=info msg="Peer Up" Key=172.16.100.3 State=BGP_FSM_OPENCONFIRM Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2073075Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012} state:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012 session_state:ESTABLISHED router_id:\"172.16.100.2\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:33577}}"
2023-06-07T07:36:34.2073777Z === RUN   Test_NeighborAddDel/update_both_neighbors
2023-06-07T07:36:34.2074628Z level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
2023-06-07T07:36:34.2075524Z level=info msg="Reconciling peers for virtual router with local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2076381Z level=info msg="Updating peer 172.16.100.2/32 65011 in local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2077272Z level=info msg="Update timer configuration" Err="<nil>" Key=172.16.100.2 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2078035Z level=info msg="Resetting peer 172.16.100.2 (ASN 65011) due to a config change" subsys=bgp-control-plane
2023-06-07T07:36:34.2078904Z level=info msg="Send operation notification" Key=172.16.100.2 Op="Neighbor reset" Topic=Operation asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2079783Z level=info msg="Updating peer 172.16.100.3/32 65012 in local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2080652Z level=info msg="Update timer configuration" Err="<nil>" Key=172.16.100.3 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2081421Z level=info msg="Resetting peer 172.16.100.3 (ASN 65012) due to a config change" subsys=bgp-control-plane
2023-06-07T07:36:34.2082320Z level=info msg="Send operation notification" Key=172.16.100.3 Op="Neighbor reset" Topic=Operation asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2083243Z level=info msg="Done reconciling peers for virtual router with local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2084412Z level=warning msg="sent notification" Code=6 Communicated-Reason="Peer configuration changed" Data="[]" Key=172.16.100.2 State=BGP_FSM_ESTABLISHED Subcode=4 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2085602Z time="2023-06-07T07:35:15Z" level=warning msg="received notification" Code=6 Communicated-Reason="Peer configuration changed" Data="[]" Key=172.16.100.1 Subcode=4 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2086818Z level=warning msg="sent notification" Code=6 Communicated-Reason="Peer configuration changed" Data="[]" Key=172.16.100.3 State=BGP_FSM_ESTABLISHED Subcode=4 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2088025Z time="2023-06-07T07:35:15Z" level=info msg="Peer Down" Key=172.16.100.1 Reason="notification-received code 6(cease) subcode 4(administrative reset)" State=BGP_FSM_ESTABLISHED Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2089236Z level=info msg="Peer Down" Key=172.16.100.2 Reason="notification-sent code 6(cease) subcode 4(administrative reset)" State=BGP_FSM_ESTABLISHED Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2090480Z time="2023-06-07T07:35:15Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_IDLE old=BGP_FSM_ESTABLISHED reason="notification-received code 6(cease) subcode 4(administrative reset)" subsys=basic
2023-06-07T07:36:34.2091853Z level=info msg="Peer Down" Key=172.16.100.3 Reason="notification-sent code 6(cease) subcode 4(administrative reset)" State=BGP_FSM_ESTABLISHED Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2092690Z time="2023-06-07T07:35:15Z" level=info msg="GoBGP test instance: Peer Event: {65001 IDLE}"
2023-06-07T07:36:34.2093630Z time="2023-06-07T07:35:15Z" level=warning msg="received notification" Code=6 Communicated-Reason="Peer configuration changed" Data="[]" Key=172.16.100.1 Subcode=4 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2094902Z time="2023-06-07T07:35:15Z" level=info msg="Peer Down" Key=172.16.100.1 Reason="notification-received code 6(cease) subcode 4(administrative reset)" State=BGP_FSM_ESTABLISHED Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2095887Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011} state:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011 session_state:IDLE router_id:\"172.16.100.2\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:46807}}"
2023-06-07T07:36:34.2097156Z time="2023-06-07T07:35:15Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_IDLE old=BGP_FSM_ESTABLISHED reason="notification-received code 6(cease) subcode 4(administrative reset)" subsys=basic
2023-06-07T07:36:34.2098267Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012} state:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012 session_state:IDLE router_id:\"172.16.100.2\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:33577}}"
2023-06-07T07:36:34.2099294Z time="2023-06-07T07:35:20Z" level=debug msg="IdleHoldTimer expired" Duration=5 Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2100134Z time="2023-06-07T07:35:20Z" level=debug msg="IdleHoldTimer expired" Duration=5 Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2101093Z time="2023-06-07T07:35:20Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_ACTIVE old=BGP_FSM_IDLE reason=idle-hold-timer-expired subsys=basic
2023-06-07T07:36:34.2101870Z time="2023-06-07T07:35:20Z" level=info msg="GoBGP test instance: Peer Event: {65001 ACTIVE}"
2023-06-07T07:36:34.2123038Z time="2023-06-07T07:35:20Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_ACTIVE old=BGP_FSM_IDLE reason=idle-hold-timer-expired subsys=basic
2023-06-07T07:36:34.2123994Z time="2023-06-07T07:35:28Z" level=debug msg="try to connect" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2124776Z time="2023-06-07T07:35:28Z" level=debug msg="try to connect" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2125679Z time="2023-06-07T07:35:28Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_OPENSENT old=BGP_FSM_ACTIVE reason=new-connection subsys=basic
2023-06-07T07:36:34.2126462Z time="2023-06-07T07:35:28Z" level=info msg="GoBGP test instance: Peer Event: {65001 OPENSENT}"
2023-06-07T07:36:34.2127337Z time="2023-06-07T07:35:28Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_OPENSENT old=BGP_FSM_ACTIVE reason=new-connection subsys=basic
2023-06-07T07:36:34.2128314Z level=warning msg="Closed an accepted connection" Key=172.16.100.2 State=BGP_FSM_IDLE Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2129286Z level=warning msg="Closed an accepted connection" Key=172.16.100.3 State=BGP_FSM_IDLE Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2130492Z time="2023-06-07T07:35:28Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_IDLE old=BGP_FSM_OPENSENT reason=read-failed subsys=basic
2023-06-07T07:36:34.2131223Z time="2023-06-07T07:35:28Z" level=info msg="GoBGP test instance: Peer Event: {65001 IDLE}"
2023-06-07T07:36:34.2132090Z time="2023-06-07T07:35:28Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_IDLE old=BGP_FSM_OPENSENT reason=read-failed subsys=basic
2023-06-07T07:36:34.2132979Z time="2023-06-07T07:35:33Z" level=debug msg="IdleHoldTimer expired" Duration=5 Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2134081Z time="2023-06-07T07:35:33Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_ACTIVE old=BGP_FSM_IDLE reason=idle-hold-timer-expired subsys=basic
2023-06-07T07:36:34.2135019Z time="2023-06-07T07:35:33Z" level=debug msg="IdleHoldTimer expired" Duration=5 Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2135975Z time="2023-06-07T07:35:33Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_ACTIVE old=BGP_FSM_IDLE reason=idle-hold-timer-expired subsys=basic
2023-06-07T07:36:34.2136722Z time="2023-06-07T07:35:33Z" level=info msg="GoBGP test instance: Peer Event: {65001 ACTIVE}"
2023-06-07T07:36:34.2137428Z time="2023-06-07T07:35:41Z" level=debug msg="try to connect" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2138504Z time="2023-06-07T07:35:41Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_OPENSENT old=BGP_FSM_ACTIVE reason=new-connection subsys=basic
2023-06-07T07:36:34.2139490Z level=warning msg="Closed an accepted connection" Key=172.16.100.3 State=BGP_FSM_IDLE Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2140475Z time="2023-06-07T07:35:41Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_IDLE old=BGP_FSM_OPENSENT reason=read-failed subsys=basic
2023-06-07T07:36:34.2141287Z time="2023-06-07T07:35:42Z" level=debug msg="try to connect" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2142223Z time="2023-06-07T07:35:42Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_OPENSENT old=BGP_FSM_ACTIVE reason=new-connection subsys=basic
2023-06-07T07:36:34.2143259Z level=warning msg="Closed an accepted connection" Key=172.16.100.2 State=BGP_FSM_IDLE Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2144007Z time="2023-06-07T07:35:42Z" level=info msg="GoBGP test instance: Peer Event: {65001 OPENSENT}"
2023-06-07T07:36:34.2144863Z time="2023-06-07T07:35:42Z" level=debug msg="state changed" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP new=BGP_FSM_IDLE old=BGP_FSM_OPENSENT reason=read-failed subsys=basic
2023-06-07T07:36:34.2145569Z time="2023-06-07T07:35:42Z" level=info msg="GoBGP test instance: Peer Event: {65001 IDLE}"
2023-06-07T07:36:34.2146321Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011} state:{local_asn:65001 neighbor_address:\"172.16.100.2\" peer_asn:65011 session_state:ACTIVE router_id:\"172.16.100.2\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:46807}}"
2023-06-07T07:36:34.2147330Z level=info msg="type:STATE peer:{conf:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012} state:{local_asn:65001 neighbor_address:\"172.16.100.3\" peer_asn:65012 session_state:ACTIVE router_id:\"172.16.100.2\"} transport:{local_address:\"172.16.100.1\" local_port:1790 remote_port:33577}}"
2023-06-07T07:36:34.2147967Z     neighbor_test.go:149: 
2023-06-07T07:36:34.2148508Z         	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:149
2023-06-07T07:36:34.2148905Z         	Error:      	Received unexpected error:
2023-06-07T07:36:34.2149353Z         	            	did not receive expected peering state ["ESTABLISHED"], context deadline exceeded
2023-06-07T07:36:34.2149813Z         	Test:       	Test_NeighborAddDel/update_both_neighbors
2023-06-07T07:36:34.2150179Z         	Messages:   	update both neighbors
2023-06-07T07:36:34.2150543Z === RUN   Test_NeighborAddDel/delete_both_neighbors
2023-06-07T07:36:34.2150882Z     neighbor_test.go:149: 
2023-06-07T07:36:34.2151237Z         	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:149
2023-06-07T07:36:34.2151629Z         	Error:      	Received unexpected error:
2023-06-07T07:36:34.2152207Z         	            	did not receive expected peering state ["IDLE" "ACTIVE"], context deadline exceeded
2023-06-07T07:36:34.2152600Z         	Test:       	Test_NeighborAddDel/delete_both_neighbors
2023-06-07T07:36:34.2152913Z         	Messages:   	delete both neighbors
2023-06-07T07:36:34.2153358Z time="2023-06-07T07:35:46Z" level=info msg="GoBGP test instance: stopping"
2023-06-07T07:36:34.2153946Z time="2023-06-07T07:35:46Z" level=info msg="Delete a peer configuration" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2154601Z level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
2023-06-07T07:36:34.2155272Z time="2023-06-07T07:35:46Z" level=debug msg="freed fsm.h" Key=172.16.100.1 State=BGP_FSM_IDLE Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2155781Z time="2023-06-07T07:35:46Z" level=info msg="GoBGP test instance: stopping"
2023-06-07T07:36:34.2156358Z time="2023-06-07T07:35:46Z" level=info msg="Delete a peer configuration" Key=172.16.100.1 Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2157017Z level=info msg="Reconciling peers for virtual router with local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2157693Z time="2023-06-07T07:35:46Z" level=debug msg="freed fsm.h" Key=172.16.100.1 State=BGP_FSM_IDLE Topic=Peer asn=65011 component=tests.BGP subsys=basic
2023-06-07T07:36:34.2158336Z level=info msg="Removing peer 172.16.100.2/32 65011 from local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2159000Z level=info msg="Delete a peer configuration" Key=172.16.100.2 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2159655Z level=info msg="Removing peer 172.16.100.3/32 65012 from local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2160328Z level=info msg="Delete a peer configuration" Key=172.16.100.3 Topic=Peer asn=65001 component=gobgp.BgpServerInstance subsys=bgp-control-plane
2023-06-07T07:36:34.2161023Z level=info msg="Done reconciling peers for virtual router with local ASN 65001" component=manager.neighborReconciler subsys=bgp-control-plane
2023-06-07T07:36:34.2161451Z level=info msg=Stopping subsys=hive
2023-06-07T07:36:34.2161917Z time="2023-06-07T07:35:46Z" level=info msg="deleting dummy links"
2023-06-07T07:36:34.2162286Z --- FAIL: Test_NeighborAddDel (40.05s)
2023-06-07T07:36:34.2162687Z     --- PASS: Test_NeighborAddDel/add_two_neighbors (8.70s)
2023-06-07T07:36:34.2163115Z     --- FAIL: Test_NeighborAddDel/update_both_neighbors (30.99s)
2023-06-07T07:36:34.2163553Z     --- FAIL: Test_NeighborAddDel/delete_both_neighbors (0.00s)
```

--- a/pkg/bgpv1/test/neighbor_test.go
+++ b/pkg/bgpv1/test/neighbor_test.go
@@ -21,8 +21,8 @@ import (
 
 var (
 	// maxNeighborTestDuration is allowed time for the Neighbor tests execution
-	// (on average we need about 20 - 25s to finish the tests due to BGP timers etc.)
-	maxNeighborTestDuration = 40 * time.Second
+	// (on average we need about 35-40s to finish the tests due to BGP timers etc.)
+	maxNeighborTestDuration = 1 * time.Minute
 )
 
 // peeringState helper struct containing peering information of BGP neighbor

